b_cfn_sagemaker_endpoint/resource.py

```diff
@@ -0,0 +1,118 @@
+from typing import Iterable
+
+from aws_cdk.aws_s3 import Bucket, EventType
+from aws_cdk.aws_sagemaker import CfnEndpointConfigProps, CfnEndpointProps, CfnEndpoint, CfnEndpointConfig
+from aws_cdk.core import Construct
+
+from b_cfn_sagemaker_endpoint.bucket_event import BucketEvent
+from b_cfn_sagemaker_endpoint.model_props import ModelProps
+from b_cfn_sagemaker_endpoint.refresh.function import RefreshFunction
+
+
+class SagemakerEndpoint(Construct):
+    """
+    SageMaker model inference endpoint.
+
+    This resource enables deployed endpoint to be automatically updated
+    when new model(-s) data is uploaded.
+
+    :param scope: Construct scope.
+    :param id: Scoped id of the resource.
+    :param endpoint_props: Properties for defining a ``AWS::SageMaker::Endpoint``.
+    :param endpoint_config_props: Properties for defining a ``AWS::SageMaker::EndpointConfig``.
+    :param models_props: SageMaker models properties.
+    :param models_bucket: Source S3 bucket for models data.
+    :param bucket_events: Models data bucket events. By default, all `models_bucket`
+        objects events (`OBJECT_CREATED`) are handled.
+    :param wait_time: Time to wait before endpoint is updated. It is useful to wait before
+        handling s3 bucket events as there can be multiple other in-flight events coming.
+        Default is 60 seconds.
+    """
+
+    def __init__(
+            self,
+            scope: Construct,
+            id: str,
+            endpoint_props: CfnEndpointProps,
+            endpoint_config_props: CfnEndpointConfigProps,
+            models_props: Iterable[ModelProps],
+            models_bucket: Bucket,
+            bucket_events: Iterable[BucketEvent] = None,
+            wait_time: float = 60
+    ):
+        if endpoint_props.endpoint_config_name != endpoint_config_props.endpoint_config_name:
+            raise ValueError(
+                'Endpoint config name setting must be identical in the '
+                '`endpoint_props` and the `endpoint_config_props` properties.'
+            )
+
+        if bucket_events is None:
+            bucket_events = [
+                BucketEvent(EventType.OBJECT_CREATED)
+            ]
+
+        super().__init__(scope, id)
+
+        models = [props.bind(self) for props in models_props]
+        endpoint_config_a = CfnEndpointConfig(
```

Comment:
Explain clearly A and B - what are they.

Reply:
done. Documented about it in README and in function class